### PR TITLE
feat: add forceTurnOnSoftwareKeyboard capability

### PR DESCRIPTION
**Pull request by KazuCocoa**
To make https://github.com/appium/WebDriverAgent/pull/524 configurable via capabilities.
This Pr needs to bump WDA dependency after https://github.com/appium/WebDriverAgent/pull/524 merge

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -166,7 +166,7 @@ Capability | Description
...
 |`appium:connectHardwareKeyboard`|Set this option to `true` in order to enable hardware keyboard in Simulator. The preference works only when Appium launches a simulator instance with this value. It is set to `false` by default, because this helps to workaround some XCTest bugs.|`true` or `false`|
-|`appium:forceTurnOnSoftwareKeyboard`|Set this option to `true` in order to turn software keyboard on since Appium 1.22.0. This option helps to avoid `Keyboard is not present` error. It is set to `true` by default. Appium respect preset simulator software/hardware keyboard preference when this value is `false`. |`true` or `false`|
+|`appium:forceTurnOnSoftwareKeyboard`|Set this option to `true` in order to turn software keyboard on since Appium 1.22.0. This option helps to avoid `Keyboard is not present` error. It is set to `true` by default. Appium respect preset simulator software/hardware keyboard preference when this value is `false`. This preference is prior rather than `connectHardwareKeyboard`. |`true` or `false`|
```
Comment (mykola-mokhnach) [inline README.md:169]: This preference has priority over ...

File: README.md [hunk 2]
```diff
@@ -165,7 +165,8 @@ Capability | Description
...
-|`appium:connectHardwareKeyboard`|Set this option to `true` in order to enable hardware keyboard in Simulator. It is set to `false` by default, because this helps to workaround some XCTest bugs.|`true` or `false`|
+|`appium:connectHardwareKeyboard`|Set this option to `true` in order to enable hardware keyboard in Simulator. The preference works only when Appium launches a simulator instance with this value. It is set to `false` by default, because this helps to workaround some XCTest bugs.|`true` or `false`|
+|`appium:forceTurnOnSoftwareKeyboard`|Set this option to `true` in order to turn software keyboard on since Appium 1.22.0. This option helps to avoid `Keyboard is not present` error. It is set to `true` by default. Appium respect preset simulator software/hardware keyboard preference when this value is `false`. This preference is prior rather than `connectHardwareKeyboard`. |`true` or `false`|
```
Comment (mykola-mokhnach) [inline README.md:169]: I'm still not fully convinced on why we need two capabilities here, while we could only keep `connectHardwareKeyboard `

File: README.md (hunk 1 above)
Comment (mykola-mokhnach) [reply README.md:169]: Please also describe the point of having two capabilities (you could just copy https://github.com/appium/WebDriverAgent/pull/524#discussion_r689047619)

File: lib/driver.js [hunk 3]
```diff
@@ -986,6 +986,7 @@ class XCUITestDriver extends BaseDriver {
       shouldTerminateApp: this.opts.shouldTerminateApp ?? true,
       forceAppLaunch: this.opts.forceAppLaunch ?? true,
       useNativeCachingStrategy: this.opts.useNativeCachingStrategy ?? true,
+      forceTurnOnSoftwareKeyboard: this.opts.forceTurnOnSoftwareKeyboard ?? true,
```
Comment (mykola-mokhnach) [inline lib/driver.js:989]: would the default still work as expected is connectHardwareKeyboard is set to false?

Comment (KazuCocoa) [reply lib/driver.js]: Yes, this caps works when creating a new session. the hardware keyboard is when preparing a simulator instance. This capability set the software keyboard when I tested on my local.

File: README.md (hunk 1 above)
Comment (KazuCocoa) [reply README.md]: note: will update here later

File: README.md [hunk 4]
```diff
@@ -165,7 +165,8 @@ Capability | Description
...
-|`appium:connectHardwareKeyboard`|Set this option to `true` in order to enable hardware keyboard in Simulator. It is set to `false` by default, because this helps to workaround some XCTest bugs.|`true` or `false`|
+|`appium:connectHardwareKeyboard`|Set this option to `true` in order to enable hardware keyboard in Simulator. The preference works only when Appium launches a simulator instance with this value. It is set to `false` by default, because this helps to workaround some XCTest bugs.|`true` or `false`|
+|`appium:forceSimulatorSoftwareKeyboardPresence`|Set this option to `true` in order to turn software keyboard on and turn hardware keyboard off in Simulator since Appium 1.22.0. This option helps to avoid `Keyboard is not present` error. It is set to `true` by default. Appium respects preset simulator software/hardware keyboard preference when this value is `false`, so `connectHardwareKeyboard: false` and `forceSimulatorSoftwareKeyboardPresence: false` means Appium keeps preset simulator keyboard preference. This option has priority over `connectHardwareKeyboard`. |`true` or `false`|
```
Comment (mykola-mokhnach) [inline README.md:169]: ..means for Appium to keep the current Simulator keyboard preferences

File: lib/driver.js [hunk 5]
```diff
@@ -986,6 +986,7 @@ class XCUITestDriver extends BaseDriver {
       shouldTerminateApp: this.opts.shouldTerminateApp ?? true,
       forceAppLaunch: this.opts.forceAppLaunch ?? true,
       useNativeCachingStrategy: this.opts.useNativeCachingStrategy ?? true,
+      forceSimulatorSoftwareKeyboardPresence: this.opts.forceSimulatorSoftwareKeyboardPresence ?? true,
```
Comment (mykola-mokhnach) [inline lib/driver.js:989]: how about 
```
forceSimulatorSoftwareKeyboardPresence: this.opts.forceSimulatorSoftwareKeyboardPresence 
  ?? (this.opts.connectHardwareKeyboard === true ? false : true) ,
```

Comment (KazuCocoa) [reply lib/driver.js:989]: ok, no objection to it

File: README.md [hunk 6]
```diff
@@ -165,7 +165,8 @@ Capability | Description
...
-|`appium:connectHardwareKeyboard`|Set this option to `true` in order to enable hardware keyboard in Simulator. It is set to `false` by default, because this helps to workaround some XCTest bugs.|`true` or `false`|
+|`appium:connectHardwareKeyboard`|Set this option to `true` in order to enable hardware keyboard in Simulator. The preference works only when Appium launches a simulator instance with this value. It is set to `false` by default, because this helps to workaround some XCTest bugs. `connectHardwareKeyboard: true` makes `forceSimulatorSoftwareKeyboardPresence: false` if no `forceSimulatorSoftwareKeyboardPresence` in the capability since Appium 1.22.0. |`true` or `false`|
```
Comment (mykola-mokhnach) [inline README.md:168]: ... if no explicit value is set for `forceSimulatorSoftwareKeyboardPresence` capability ...

**Review by mykola-mokhnach — APPROVED** [review]

**Pull request merged**